vir/src/data.rs

```diff
@@ -85,6 +85,7 @@ pub enum ConstData {
     Int(u128), // TODO: what about negative numbers? larger numbers?
     Wildcard,
     Null,
+    Result,
```

Comment:
I don't think `result` belongs into constants. It is a literal, if anything. Let's put it into a separate `ExprKind`.